Fix errors like

block.h:344:24: error: 'struct bio' has no member named 'bi_sector'

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Upstream-Status: Pending

Index: git/instrumentation/events/lttng-module/block.h
===================================================================
--- git.orig/instrumentation/events/lttng-module/block.h	2014-05-02 22:20:01.814125429 -0700
+++ git/instrumentation/events/lttng-module/block.h	2014-05-02 22:28:48.927455709 -0700
@@ -341,9 +341,9 @@
 	TP_fast_assign(
 		tp_assign(dev, bio->bi_bdev ?
 					  bio->bi_bdev->bd_dev : 0)
-		tp_assign(sector, bio->bi_sector)
-		tp_assign(nr_sector, bio->bi_size >> 9)
-		blk_fill_rwbs(rwbs, bio->bi_rw, bio->bi_size)
+		tp_assign(sector, bio->bi_iter.bi_sector)
+		tp_assign(nr_sector, bio->bi_iter.bi_size >> 9)
+		blk_fill_rwbs(rwbs, bio->bi_rw, bio->bi_iter.bi_size)
 		tp_memcpy(comm, current->comm, TASK_COMM_LEN)
 	),
 
@@ -385,14 +385,14 @@
 
 	TP_fast_assign(
 		tp_assign(dev, bio->bi_bdev->bd_dev)
-		tp_assign(sector, bio->bi_sector)
-		tp_assign(nr_sector, bio->bi_size >> 9)
+		tp_assign(sector, bio->bi_iter.bi_sector)
+		tp_assign(nr_sector, bio->bi_iter.bi_size >> 9)
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,38))
 		tp_assign(error, error)
 #else
 		tp_assign(error, 0)
 #endif
-		blk_fill_rwbs(rwbs, bio->bi_rw, bio->bi_size)
+		blk_fill_rwbs(rwbs, bio->bi_rw, bio->bi_iter.bi_size)
 	),
 
 	TP_printk("%d,%d %s %llu + %u [%d]",
@@ -419,9 +419,9 @@
 
 	TP_fast_assign(
 		tp_assign(dev, bio->bi_bdev->bd_dev)
-		tp_assign(sector, bio->bi_sector)
-		tp_assign(nr_sector, bio->bi_size >> 9)
-		blk_fill_rwbs(rwbs, bio->bi_rw, bio->bi_size)
+		tp_assign(sector, bio->bi_iter.bi_sector)
+		tp_assign(nr_sector, bio->bi_iter.bi_size >> 9)
+		blk_fill_rwbs(rwbs, bio->bi_rw, bio->bi_iter.bi_size)
 		tp_memcpy(comm, current->comm, TASK_COMM_LEN)
 	),
 
@@ -485,9 +485,9 @@
 
 	TP_fast_assign(
 		tp_assign(dev, bio->bi_bdev->bd_dev)
-		tp_assign(sector, bio->bi_sector)
-		tp_assign(nr_sector, bio->bi_size >> 9)
-		blk_fill_rwbs(rwbs, bio->bi_rw, bio->bi_size)
+		tp_assign(sector, bio->bi_iter.bi_sector)
+		tp_assign(nr_sector, bio->bi_iter.bi_size >> 9)
+		blk_fill_rwbs(rwbs, bio->bi_rw, bio->bi_iter.bi_size)
 		tp_memcpy(comm, current->comm, TASK_COMM_LEN)
 	),
 
@@ -513,9 +513,9 @@
 
 	TP_fast_assign(
 		tp_assign(dev, bio->bi_bdev ? bio->bi_bdev->bd_dev : 0)
-		tp_assign(sector, bio->bi_sector)
-		tp_assign(nr_sector, bio->bi_size >> 9)
-		blk_fill_rwbs(rwbs, bio->bi_rw, bio->bi_size)
+		tp_assign(sector, bio->bi_iter.bi_sector)
+		tp_assign(nr_sector, bio->bi_iter.bi_size >> 9)
+		blk_fill_rwbs(rwbs, bio->bi_rw, bio->bi_iter.bi_size)
 		tp_memcpy(comm, current->comm, TASK_COMM_LEN)
 	),
 
@@ -587,10 +587,10 @@
 
 	TP_fast_assign(
 		tp_assign(dev, bio ? bio->bi_bdev->bd_dev : 0)
-		tp_assign(sector, bio ? bio->bi_sector : 0)
-		tp_assign(nr_sector, bio ? bio->bi_size >> 9 : 0)
+		tp_assign(sector, bio ? bio->bi_iter.bi_sector : 0)
+		tp_assign(nr_sector, bio ? bio->bi_iter.bi_size >> 9 : 0)
 		blk_fill_rwbs(rwbs, bio ? bio->bi_rw : 0,
-			      bio ? bio->bi_size >> 9 : 0)
+			      bio ? bio->bi_iter.bi_size >> 9 : 0)
 		tp_memcpy(comm, current->comm, TASK_COMM_LEN)
         ),
 
@@ -759,9 +759,9 @@
 
 	TP_fast_assign(
 		tp_assign(dev, bio->bi_bdev->bd_dev)
-		tp_assign(sector, bio->bi_sector)
+		tp_assign(sector, bio->bi_iter.bi_sector)
 		tp_assign(new_sector, new_sector)
-		blk_fill_rwbs(rwbs, bio->bi_rw, bio->bi_size)
+		blk_fill_rwbs(rwbs, bio->bi_rw, bio->bi_iter.bi_size)
 		tp_memcpy(comm, current->comm, TASK_COMM_LEN)
 	),
 
@@ -805,11 +805,11 @@
 
 	TP_fast_assign(
 		tp_assign(dev, bio->bi_bdev->bd_dev)
-		tp_assign(sector, bio->bi_sector)
-		tp_assign(nr_sector, bio->bi_size >> 9)
+		tp_assign(sector, bio->bi_iter.bi_sector)
+		tp_assign(nr_sector, bio->bi_iter.bi_size >> 9)
 		tp_assign(old_dev, dev)
 		tp_assign(old_sector, from)
-		blk_fill_rwbs(rwbs, bio->bi_rw, bio->bi_size)
+		blk_fill_rwbs(rwbs, bio->bi_rw, bio->bi_iter.bi_size)
 	),
 
 	TP_printk("%d,%d %s %llu + %u <- (%d,%d) %llu",
